src/Workspaces/Core/Portable/Shared/Extensions/ICodeDefinitionFactoryExtensions.cs

```diff
@@ -24,6 +34,17 @@ internal static partial class ICodeDefinitionFactoryExtensions
                     SpecializedCollections.EmptyList<SyntaxNode>()));
```

Comment:
so i'd prefer that we just change this line to:

```c#
             return codeDefinitionFactory.ThrowStatement(
                 codeDefinitionFactory.ObjectCreationExpression(
                      codeDefinitionFactory.TypeExpression(compilation.NotImplementedExceptionType(), addImport: false),
                      SpecializedCollections.EmptyList<SyntaxNode>()));
```

We would then create an overload of SyntaxGenerator.TypeExpression that took this new "bool addImport" parameter.  That way anyone could opt out of this default behavior if they wanted.

By doing this we still generate the right code in the presence (or absense) of "using System;".  We don't need to do any detection if "using System" is there.  And we'll have all the right logi around generating things like "global::System.NotImplementedException" (or aliases, or whatever), that we have today.